account/src/main/java/com/hms/lib/commonmobileservices/account/huawei/HuaweiAccountServiceImpl.kt

```diff
@@ -98,14 +100,44 @@ internal class HuaweiAccountServiceImpl(context: Context, signInParams: SignInPa
 
     override fun getEmail(): String? {
         val email = sharedPrefHelper.getEmail()
-        return if(email.isEmpty()){
+        return if (email.isEmpty()) {
             null
-        }
-        else{
+        } else {
             email
         }
     }
 
+    override fun startSmsRetriver(context: Context): Work<Unit> {
+        val worker: Work<Unit> = Work()
+
+        ReadSmsManager.start(context)
+            .addOnSuccessListener { worker.onSuccess(Unit) }
```

Comment:
This seems you need to create a broadcast receiver to get SMS content. This usage is not correct. You can leave this feature to the new sprint.